lib/lacework.bash

```diff
@@ -0,0 +1,276 @@
+            echo "ERROR: Missing config related to vulnerability scans. Need the following: ACCESS_TOKEN_ENV_VAR, VULNERABILITY_SCAN_REPOSITORY, VULNERABILITY_SCAN_TAG" >&2
+            exit 1
+        fi
+    fi
+
+    export FAIL_LEVEL="${BUILDKITE_PLUGIN_LACEWORK_FAIL_LEVEL:-}"
+
+}
+
+# Lacework Scans based on the provided scan tool
+function lacework_scan() {
+    case "${SCAN_TYPE}" in
+        sca)
+        lacework_sca
+        ;;
+
+        iac)
+        lacework_iac
+        ;;
+
+        vulnerability)
+        lacework_vulnerability
+        ;;
+
+        sast)
+        lacework_sast
+        ;;
+    esac
+}
+
+
+function lacework_sca() {
+
+    echo "--- Running Lacework SCA scan"
+
+    CMD=(
+    lacework
+    )
+
+    if [ -n "$PROFILE" ]; then
+        CMD+=(
+            --profile "$PROFILE"
+        )
+    else
+        CMD+=(
+        --account "${ACCOUNT_NAME}"
+        --api_key "${API_KEY_ENV_VAR}"
+        --api_secret "${API_KEY_SECRET_ENV_VAR}"
+    )
+    fi  
+
+    CMD+=(
+        sca scan .
+        --save-results
+    )
+        
+
+    echo "${CMD[@]}"
+
+    "${CMD[@]}" 
+}
+
+function lacework_sast() {
+
+    echo "--- Running Lacework SAST scan"
+
+    CMD=(
+    lacework
+    )
+
+    if [ -n "$PROFILE" ]; then
+        CMD+=(
+            --profile "$PROFILE"
+        )
+    else
+        CMD+=(
+        --account "${ACCOUNT_NAME}"
+        --api_key "${API_KEY_ENV_VAR}"
+        --api_secret "${API_KEY_SECRET_ENV_VAR}"
+        )
+    fi  
+
+    
+
+    SARIF_ARTIFACT="lacework-sast-report-${BUILDKITE_PIPELINE_SLUG}-${BUILDKITE_BUILD_NUMBER}.sarif"
+
+    CMD+=(
+        sast scan -o "${SARIF_ARTIFACT}"
+    )
+        
+
+    echo "${CMD[@]}"
+
+    "${CMD[@]}" 
+
+    annotate_and_upload_build_sast "${SARIF_ARTIFACT}"
+}
+
+function lacework_iac() {
+
+    echo "--- Running Lacework IAC scan"
+
+    CMD=(
+    lacework
+    )
+
+
+    CMD+=(
+        iac
+        "${IAC_SCAN_TYPE}"
+    )
+
+    if [ -n "$PROFILE" ]; then
+    CMD+=(
+        --iac-profile "$PROFILE"
+    )
+    fi  
+
+    if [ -n "$FAIL_LEVEL" ]; then
+    CMD+=(
+        --fail "$FAIL_LEVEL"
+    )
+    fi  
+
+
+    echo "${CMD[@]}"
+
+    "${CMD[@]}" 
+}
+
+function lacework_vulnerability() {
+
+    echo "--- Running Lacework Container Vulnerability scan"
+
+    CMD=(
+    lacework
+    )
+
+    if [ -n "$PROFILE" ]; then
+        CMD+=(
+            --profile "$PROFILE"
+        )
+    else
+        CMD+=(
+            --account-name "${ACCOUNT_NAME}"
+        )
+    fi  
+
+    #access token can't be passed via profile so we still need to pass it below
+    CMD+=(
+        --access-token "${ACCESS_TOKEN_ENV_VAR}"
+    )
+
+    CMD+=(
+        vuln-scanner -s image evaluate
+        "${VULNERABILITY_SCAN_REPOSITORY}"  
+        "${VULNERABILITY_SCAN_TAG}"
+    )
+
+    if [ -n "$FAIL_LEVEL" ]; then
+    CMD+=(
+        "--policy --$FAIL_LEVEL-violation-exit-code 1"
+    )
+    fi  
+
+    
+    echo "${CMD[@]}"
```

Comment:
```suggestion

```
Although the Build Logs will redact the Secrets matching certain patterns, we shouldn't expose information in first place.